docs/implementation.md

```diff
@@ -0,0 +1,210 @@
+## Implementing Links
+
+Before implementing your own Link, be sure to read the [usage](usage.md) document.
+
+Links implement a `request` method that takes an operation and returns the results in an Observable.
+The apollo-link's `Observable` follows the ECMAScript [proposal](https://github.com/tc39/proposal-observable#api) with some extensions provided by [zen-observable](https://github.com/zenparsing/zen-observable#api).
+
+### Creating a Simple Http Link
+
+All links extend the base class, `ApolloLink`, which contains the methods for composition as well as the signature of request.
```

Comment:
None of the composition methods inherited from `ApolloLink` seem particularly coupled to the implementation of each specific Link class.  I wonder if it might be cleaner to Make `ApolloLink` a pure interface, and move the composition helpers out to a separate object.  You're already moving in that direction with static helpers like `ApolloLink.from`